### PR TITLE
Fix #3103: Segregate OptionsFragment tests

### LINKED ISSUE #3103: Test Segregation in OptionsFragmentTest
Currently in `OptionsFragmentTest` tests are working across multiple activities but thats not correct.
Each file should contain tests only related to that file and should not check functionalities of other activities.

Segregate tests of OptionsFragmentTest into 
- OptionsFragmentTest
- AudioLanguageFragmentTest
- AppLanguageFragmentTest

Also we should add more tests to make sure that we are able to hit 100% code coverage for these files which means that if we combine all tests than they should have traversed/used each line written for the functionality.

### PR DESCRIPTION
**Pull request by Sparsh1212**
## Explanation

Fixes #3103: Tests Segregation in OptionsFragmentTest

Espresso Test results snapshots:

image
image
image
image
image
image
image
image

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by Sparsh1212** [comment]
@rt4914 
I have completed the Part-A of the issue which involves:
1. Write a test in OptionsFragmentTest to check that the correct default Language is displayed in the recycler view of the Options file. (Repeat this a total of three times to check for App Language, Audio Language, and ReadingTextSize)
2. Repeat the above three tests for a configuration change.
3. Write a test for clicking an item in the Options Recycler View and check if it is sending the correct intent based on the default Language (Repeat this three times to check for App Language, Audio Language, and ReadingTextSize)
4. Also, repeat the above three tests (Point number three) for a configuration change.

Will proceed to Part-B of the issue after getting an initial pass on Part-A.

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: Rename to `openOptionsActivity_configChange_clickAudioLanguage_opensAudioLanguageActivitySuccessfully`

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: Instead of select use `click` when clicking on items in OptionsFragment

Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: Also use `open` instead of `show` when new activity is opened.

**Review by rt4914 — CHANGES_REQUESTED** [review]
The tests are failing on Roboelectric. 
Each test should pass on Roboelectric as well as Espresso. Also make sure that CI checks also pass.

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (Sparsh1212) [reply app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: Done.

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (Sparsh1212) [reply app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: Done.

**Comment by Sparsh1212** [comment]
> The tests are failing on Roboelectric.
> Each test should pass on Roboelectric as well as Espresso. Also make sure that CI checks also pass.

Apologies! will remember to check each of them before making a commit.
Update: All the checks and tests are now passing. I have updated the PR with the requested changes. PTAL.

**Comment by rt4914** [comment]
@Sparsh1212 I will review this tomorrow but meanwhile you can move to part B too.

**Comment by Sparsh1212** [comment]
@rt4914 
Here's the update about the Part-B work that I have initiated:
- All the previous tests in `AppLanguageFragmentTest` are removed and these four new tests are added keeping in mind the ideal practice (i.e not to launch the other Activity and only use the Test File Activity): 
  1. `testAppLanguage_selectedLanguageIsEnglish()`
  2. `testAppLanguage_changeConfiguration_selectedLanguageIsEnglish()`
  3. `testAppLanguage_changeLanguageToFrench_selectedLanguageIsFrench()`
  4. `testAppLanguage_changeLanguageToFrench_changeConfiguration_selectedLanguageIsFrench() `
 
- I also tried adding this test:
```kt
   @Test 
   fun testApp_Language_pressBackBtn_SendsBackCorrectInfo(){
      launch<AppLanguageActivity>(createAppLanguageActivityIntent("English")).use {
          testCoroutineDispatchers.runCurrent()
          pressBack()
          intended(hasExtra(MESSAGE_APP_LANGUAGE_ARGUMENT_KEY, "English"))
    }
   }
```
But it is failing in Roboelectric with the error that 0 intends matched.
It looks like the Back press is getting ignored totally.
I also tried many other things to click the back button or to run the `onBackPress` function but nothing seems to work. Can you please take a look at this and guide me on what I am doing wrong here or if this approach isn't correct then how can we achieve checking that it is sending back the correct info on the back press through another approach.

- I have not made any changes to `ReadingTextSizeFragmentTest` because currently there is a PR (#2929) that will propose some changes in this file, so I think it will be better to wait to rewrite this file until those changes are merged.

- Will make all the similar changes for `AudioLanguageFragmentTest` once I get a pass on the above changes and the getting back correct info on the back press problem is resolved.  

Thanks!

**Comment by rt4914** [comment]
> ```kotlin
> @Test 
>    fun testApp_Language_pressBackBtn_SendsBackCorrectInfo(){
>       launch<AppLanguageActivity>(createAppLanguageActivityIntent("English")).use {
>           testCoroutineDispatchers.runCurrent()
>           pressBack()
>           intended(hasExtra(MESSAGE_APP_LANGUAGE_ARGUMENT_KEY, "English"))
>     }
>    }
> ```

@Sparsh1212 Have you checked ShadowActivity? https://stackoverflow.com/a/8991026

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]:
```suggestion
      onView(isRoot()).perform(orientationLandscape())
      testCoroutineDispatchers.runCurrent()
```
For stabilizing  after changing orientation

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]:
```suggestion
  fun openOptionsActivity_clickReadingTextSize_opensReadingTextSizeActivity() {
```

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: can we use these original items only?
For example check `AdministratorControlsActivity` at `getIntentKey` which is used in `testNavDrawer_withAdminProfile_adminMenu_opensAdminControlsActivity`

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]:
```suggestion
  fun openOptionsActivity_configChange_clickTextSize_opensReadingTextSizeActivity() {
```

File: app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt
Comment (anandwana001) [inline app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt]:
```suggestion
```

I think we don't need `Intents.init()` and `Intents.release()` as there is no test which testing anything related to intents.

File: app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt
Comment (anandwana001) [inline app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt]:
```suggestion
  fun testAppLanguage_configChange_selectedLanguageIsEnglish() {
```

File: app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt
Comment (anandwana001) [inline app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt:112]: Let's use the similar function in the other file as well `app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt`

**Review by anandwana001 — CHANGES_REQUESTED** [review]
regarding the `testApp_Language_pressBackBtn_SendsBackCorrectInfo `, we cannot use ShadowActivity, this will make this test robolectric only. 

The reason, I am seeing here why the test fails is:

```kt
override fun onBackPressed() {
    val message = appLanguageActivityPresenter.getLanguageSelected()
    val intent = Intent()
    intent.putExtra(MESSAGE_APP_LANGUAGE_ARGUMENT_KEY, message)
    setResult(REQUEST_CODE_APP_LANGUAGE, intent)
    finish()
  }
```

on back press, the activity gets finish, so test is basically trying to test after activity gets finished. 
I don't think this test will pass on espresso also. This test seems like to come under e2e test. 

Also, add all the espresso test result screenshots in PR description.

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (Sparsh1212) [reply app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: There is no `getIntentKey` function present in AppLangaugeActivity and AudioLanguageActivity, therefore I have to again declare the constants here. Shall I add `getIntentKey` function to AppLanguageActivity in the companion object or go by the current approach? Please let me know.

**Comment by Sparsh1212** [comment]
> regarding the `testApp_Language_pressBackBtn_SendsBackCorrectInfo `, we cannot use ShadowActivity, this will make this test robolectric only.
> 
> The reason, I am seeing here why the test fails is:
> 
> ```kotlin
> override fun onBackPressed() {
>     val message = appLanguageActivityPresenter.getLanguageSelected()
>     val intent = Intent()
>     intent.putExtra(MESSAGE_APP_LANGUAGE_ARGUMENT_KEY, message)
>     setResult(REQUEST_CODE_APP_LANGUAGE, intent)
>     finish()
>   }
> ```
> 
> on back press, the activity gets finish, so test is basically trying to test after activity gets finished.
> I don't think this test will pass on espresso also. This test seems like to come under e2e test.
> 
> Also, add all the espresso test result screenshots in PR description.

Thanks, will update and make all the suggested changes.

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: introducing `getIntentKey`, will be helpful. @rt4914 could you please confirm?

Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: Actually we should shift keys to `companion object` of `OptionsFragment` and from there we can use them wherever we want using `OptionsFragment. KEY_AUDIO_LANGUAGE_PREFERENCE_SUMMARY_VALUE `

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (Sparsh1212) [reply app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: Done.

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (Sparsh1212) [reply app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: Done.

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (Sparsh1212) [reply app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: Done.

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (Sparsh1212) [reply app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: Done.

File: app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt
Comment (Sparsh1212) [reply app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt]: Done.

File: app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt
Comment (Sparsh1212) [reply app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt]: Done.

File: app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt
Comment (Sparsh1212) [reply app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt:112]: Done.

**Comment by Sparsh1212** [comment]
@rt4914 @anandwana001 I have made all the suggested changes. PTAL.

File: app/src/main/java/org/oppia/android/app/options/OptionsFragment.kt
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/options/OptionsFragment.kt]: If you are introducing these value then please remove the previous consts mentioned at the top of the kotlin file.
Or alternate can be to not introduce these consts and use the values mentioned at top of kotlin file in your test file.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@Sparsh1212 Suggested change.

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]:
```suggestion
  fun testOptionsFragment_defaultReadingTextSizeIsDisplayed() {
```

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]:
```suggestion
  fun testOptionsFragment_configChange_defaultAppLanguageIsDisplayed() {
```

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]:
```suggestion
  fun testOptionsFragment_configChange_defaultAudioLanguageIsDisplayed() {
```

Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: ditto for other test names.

**Review by anandwana001 — CHANGES_REQUESTED** [review]
1. Add Espresso Test result screenshot of all the test files that exists in this PR. 
2. In the tablet, we use Master-Detail design, so I think we also need to add a test for tablet-specific. 
```kt
  @Test
  @Config(qualifiers = "sw600dp")
fun testAppLanguage_clickAppLanguage_changeAppLanguage_checkOptionsFragmentIsUpdatedCorrectly()
```
and other edge cases too. 

3. For `app/src/test/java/org/oppia/android/app/testing/options/ReadingTextSizeFragmentTest.kt`, could you mention what all cases does #2929 covering and what cases are need to be covered as a part of this PR. This way we can keep track of all the cases to get covered in testing. 

4. Let's remove the unused variables. 
```kt
private const val NO_AUDIO = 0
private const val HINDI = 3
private const val CHINESE = 4
```

5. Add kdoc for test files 
reference :
```kt
/** Tests for [OptionsFragment]. */
@RunWith(AndroidJUnit4::class)
``` 

6. Could you confirm if we need `ReadingTextSizeActivityTest`?

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (Sparsh1212) [reply app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: Done.

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (Sparsh1212) [reply app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: Done.

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (Sparsh1212) [reply app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt]: Done.

**Comment by Sparsh1212** [comment]
@anandwana001 

Regarding:
1. I tried running the Espresso tests by right-clicking the file and running the test for `AppLanguageFragmentTest` and `AudioLanguageFragmentTest` but it is running the Roboelectric tests instead of Espresso. I also tried running them by making the Espresso configuration in the Edit Configuration but the tests are still not running, it displays `No tests found in class`. This problem isn't occurring for `OptionsFragmentTest`. Can you please cross-check if you are facing the same problem?
2. Tests added for tablet configuration.
3. It currently has the following tests:
    1. `testTextSize_checkTextSizeOfAllFourItems_textSizeMatchedCorrectly()` 
    2. `testTextSize_changeTextSizeToLarge_changeConfiguration_checkTextSizeLargeIsSelected()`
    3. `testTextSize_changeTextSizeToMedium_mediumItemIsSelected()`
    The recent changes in this PR has covered all the required tests. So, I think we don't have to make any further changes to it.
4. Removed unused variables.
5. Added kdocs.
6. Yes we need the `ReadingTextSizeActivityTest` to check that the accessibility label is defined properly. There's also a corresponding issue for this: #2817

File: app/src/main/java/org/oppia/android/app/options/OptionsFragment.kt
Comment (Sparsh1212) [reply app/src/main/java/org/oppia/android/app/options/OptionsFragment.kt]: I have removed the previous consts mentioned at the top of the test file. PTAL.

**Comment by anandwana001** [comment]
Looking into the file path, `AudioLanguageFragmentTest` and `AppLanguageFragmentTest` are always in the Local unit tests folder which is `app/testing/..`. that's why there are not running on espresso. 

If you look here there are 2 files:
1. src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
2. src/test/java/org/oppia/android/app/testing/options/OptionsFragmentTest.kt

`sharedTest` directory is used for instrumentation test and `test` directory is used for unit test, and the test we are working for in this PR is usually instrumentation test. 
You need to shift the required test to `sharedTest` directory.

**Comment by Sparsh1212** [comment]
> Looking into the file path, `AudioLanguageFragmentTest` and `AppLanguageFragmentTest` are always in the Local unit tests folder which is `app/testing/..`. that's why there are not running on espresso.
> 
> If you look here there are 2 files:
> 
> 1. src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
> 2. src/test/java/org/oppia/android/app/testing/options/OptionsFragmentTest.kt
> 
> `sharedTest` directory is used for instrumentation test and `test` directory is used for unit test, and the test we are working for in this PR is usually instrumentation test.
> You need to shift the required test to `sharedTest` directory.

Thanks a lot.
I have shifted them to `sharedTest` directory and all the Espresso tests are passing as well. 
Updated the PR description. PTAL.

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt:253]: please add espresso test result for these test too, on tablet.

File: app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageFragmentTest.kt:118]: ditto

File: app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageFragmentTest.kt:139]: ditto

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @Sparsh1212 
shifting to sharedTest is good. 

Need tablet test result screenshot. Please run these tests on tablet config and see if they are passing correctly.

**Comment by rt4914** [comment]
Un-assigning until @anandwana001 approves it.

File: app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageFragmentTest.kt
Comment (Sparsh1212) [reply app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageFragmentTest.kt:139]: Done.

File: app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageFragmentTest.kt
Comment (Sparsh1212) [reply app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageFragmentTest.kt:118]: Done.

File: app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
Comment (Sparsh1212) [reply app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt:253]: Done.

**Comment by Sparsh1212** [comment]
> Thanks @Sparsh1212
> shifting to sharedTest is good.
> 
> Need tablet test result screenshot. Please run these tests on tablet config and see if they are passing correctly.

Added.

**Review by anandwana001 — APPROVED** [review]
LGTM, Thanks @Sparsh1212

File: app/src/main/java/org/oppia/android/app/options/OptionsFragment.kt
Comment (rt4914) [reply app/src/main/java/org/oppia/android/app/options/OptionsFragment.kt]: Actually you have removed it from test file. But instead what we want it we should either use only 
```
const val MESSAGE_READING_TEXT_SIZE_ARGUMENT_KEY = "OptionsFragment.message_reading_text_size"
const val MESSAGE_APP_LANGUAGE_ARGUMENT_KEY = "OptionsFragment.message_app_language"
const val MESSAGE_AUDIO_LANGUAGE_ARGUMENT_KEY = "OptionsFragment.message_audio_language"
```
in tests and production code

or we should use only
```
const val KEY_READING_TEXT_SIZE_PREFERENCE_SUMMARY_VALUE =
      "READING_TEXT_SIZE_PREFERENCE_SUMMARY_VALUE"
    const val APP_LANGUAGE_PREFERENCE_SUMMARY_VALUE_EXTRA_KEY =
      "AppLanguageActivity.app_language_preference_summary_value"
    const val KEY_AUDIO_LANGUAGE_PREFERENCE_SUMMARY_VALUE =
      "AUDIO_LANGUAGE_PREFERENCE_SUMMARY_VALUE"
```


Currently you are using the above for the app layer code and the below for test cases which is incorrect. We should use only one of these sets in both code.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@Sparsh1212 Replied to my one of the open comments.

File: app/src/main/java/org/oppia/android/app/options/OptionsFragment.kt
Comment (Sparsh1212) [reply app/src/main/java/org/oppia/android/app/options/OptionsFragment.kt]: Actually, we are passing the key `KEY_READING_TEXT_SIZE_PREFERENCE_SUMMARY_VALUE` while creating the `ReadingTextSizeActivity` intent. In the `OptionsFragment` test, we are checking that the correct intent has passed or not when the user clicks on the `RecyclerView` item, so we need the value of this key in the test file to match the intent passed, but the problem with accessing this key in the test file is that it has been declared as an `internal const` in the companion object of `ReadingTextSizeActivity` therefore the test fails in Bazel roboelectric because of the Test file not being able to access those internal consts, therefore I have to again declare them in `companion object` of `ReadingTextSizeFragment`. Please suggested an alternative approach to this if this is incorrect so that we can access the key in the test file.

Comment (rt4914) [reply app/src/main/java/org/oppia/android/app/options/OptionsFragment.kt]: You can use the values of removing the `internal` in `ReadingTextSizeActivity` . Won't that work? @Sparsh1212 

Also the solution where you have suggested introducing them in `ReadingTextSizeFragment ` is incorrect because its not part of intent or activity so the testing would be incorrect.

Comment (Sparsh1212) [reply app/src/main/java/org/oppia/android/app/options/OptionsFragment.kt]: @rt4914  Yes removing `internal` in `ReadingTextSizeActivity` should work. Also, apologies about the alternative approach, I meant `OptionsFragment` not `ReadingTextSizeFragment`. Made a typo there. 
I have updated the changes. PTAL.
Also, the CI checks are failing, I hope this isn't due to this PR.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.
Unfortunately we cannot merge this until #3157 is fixed.

**Comment by Sparsh1212** [comment]
@rt4914 I have taken the pull from the latest develop branch. The checks are also passing. Looks like we can merge this now :)

**Pull request merged**
